lib/exhmac.ex

```diff
@@ -29,6 +29,14 @@ defmodule ExHMAC do
     }
   end
 
+  def validate_signature(signature, method, url, content, ts, secret) do
+    data = prepare_signature_data(method, url, content, ts)
+    case signature(data, secret) == signature do
+      true -> :valid
+      _ -> :invalid
+    end
```

Comment:
1. Variable should probably be renamed not to match the function name 'signature'.
2. I'd write the case statement as

```
case signature(data, secret) do
    signature -> :valid
    _ -> invalid
end
``` + renaming the variables

Reply:
Everything else looks good to me. 👌